src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -686,17 +715,11 @@ def custom_command(self, name, method_name, **kwargs):
         :rtype: None
         """
         self._check_stale()
-        merged_kwargs = self.group_kwargs.copy()
-        custom_command_type = merged_kwargs.get('custom_command_type', None)
-        if not custom_command_type:
-            raise CLIError('Module does not have `custom_command_type` set!')
-        if custom_command_type:
-            merged_kwargs.update(custom_command_type.settings)
-        merged_kwargs.update(kwargs)
-
+        merged_kwargs = self._merge_kwargs(kwargs)
+        operations_tmpl = self._resolve_kwarg(merged_kwargs, 'operations_tmpl', 'custom_command_type', required=True)
```

Comment:
same in with `command()`. Having to provide a 'default_source_name' every time looks clunky